demos/python_demos/common/images_capture.py

```diff
@@ -0,0 +1,116 @@
+import os
+import copy
+
+import cv2
+
+
+class InvalidInput(Exception):
+    pass
+
+
+class ImagesCapture:
+
+    def __init__(self, loop):
+        self.loop = loop
+
+    def read():
+        raise NotImplementedError
+
+
+class ImreadWrapper(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        self.image = cv2.imread(input, cv2.IMREAD_COLOR)
+        self.can_read = True
+        if self.image is None:
+            raise InvalidInput
+
+    def read(self):
+        if self.loop:
+            return copy.deepcopy(self.image)
+        if self.can_read:
+            self.can_read = False
+            return copy.deepcopy(self.image)
+        return None
+
+
+class DirReader(ImagesCapture):
+
+    def __init__(self, input, loop):
+        super().__init__(loop)
+        self.dir = input
+        if not os.path.isdir(self.dir):
+            raise InvalidInput
```

Comment:
Add some message to error which can explain it.

Reply:
This exception is used only for `open_images_capture()` no one is expected to see that exception.

Reply:
If the input cannot be opened (and InvalidInput catch it), the RuntimeError with message will appear at the end of `open_images_capture()`

Reply:
Done